### PR TITLE
Add rename endpoint

### PR DESCRIPTION
**Pull request by NakulManchanda**
added rename api and tests

Its based on zosmf https://www.ibm.com/support/knowledgecenter/SSLTBW_2.4.0/com.ibm.zos.v2r4.izua700/IZUHPINFO_API_PutDataSetMemberUtilities.htm

### CODE REVIEW

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunner.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    protected String getResult(ResponseCache responseCache) throws IOException {
+        String response = responseCache.getEntity();
+        return response;
+    }
+
+    @Override
+    protected ZoweApiRestException createGeneralException(ResponseCache responseCache, URI uri) throws IOException {
+        return super.createGeneralException(responseCache, uri);
+    }
```
Comment (stevenhorsman) [inline data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunner.java]: This method isn't needed if you are just overriding to call super

File: data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunnerTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        
+        String member = "OLD";
+        String withMember = String.format("%s(%s)", dsn, member);
+        String withtMemberReqString = String.format("{\"request\":\"rename\", \"from-dataset\":{\"dsn\":\"%s\", \"member\":\"%s\"} }", dsn, member);
+        zosmfRenameReq = ZosmfRenameRequest.createFromDataSetRenameRequest(withMember);
+        assertEquals(dsn, zosmfRenameReq.getDsn());
+        assertEquals(member, zosmfRenameReq.getMember());
+        assertEquals(withtMemberReqString, zosmfRenameReq.toString());
+    }
+
```
Comment (stevenhorsman) [inline data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunnerTest.java]: I'd like to see more tests reflecting the different error types you have in the runner - I don't think you have full coverage

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/AbstractDataSetsIntegrationTest.java [hunk 3]
```diff
@@ -102,4 +103,8 @@ static Response deleteDataSet(String dataSetName) {
     static String getDataSetMemberPath(String pds, String member) {
         return pds + "(" + member + ")";
     }
+    
+    static Response putDataSetRename(String oldDataSetName, DataSetRenameRequest body) {
+        return RestAssured.given().contentType("application/json").body(body).when().put(oldDataSetName + "/rename");
+    }
```
Comment (stevenhorsman) [inline data-sets-tests/src/test/java/org/zowe/data/sets/tests/AbstractDataSetsIntegrationTest.java]: This code doesn't seem to be shared, so should be pushed down into the sub class

File: data-sets-api-server/src/main/java/org/zowe/data/sets/controller/DataSetsController.java [hunk 4]
```diff
@@ -111,6 +112,16 @@
 
         return ResponseEntity.noContent().eTag(putEtag).build();
     }
+    
+    @PutMapping(value = "{oldDataSetName}/rename", produces = { "application/json" })
+    @ApiOperation(value = "Sets the content of a sequential data set, or PDS member", nickname = "renameContent", notes = "This API renames data set or partitioned data set member.", tags = "Data Sets APIs")
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/controller/DataSetsController.java]: updated value to reflect rename

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/DataSetService.java [hunk 5]
```diff
@@ -30,4 +31,6 @@
     String createDataSet(DataSetCreateRequest input);
 
     void deleteDataSet(String dataSetName);
+
+    String putRename(String oldDataSetName, DataSetRenameRequest input);
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/services/DataSetService.java]: refactored to renameDataSet

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/AbstractDataSetsIntegrationTest.java (hunk 3 above)
Comment (NakulManchanda) [reply data-sets-tests/src/test/java/org/zowe/data/sets/tests/AbstractDataSetsIntegrationTest.java]: moved down as recommended

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunner.java (hunk 1 above)
Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunner.java]: removed

File: data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java [hunk 6]
```diff
@@ -182,7 +183,25 @@ public void put_data_set_content_success() throws Exception {
...
+        mockMvc
+            .perform(put(ENDPOINT_ROOT + "/{dsn}/rename", oldName)
+            .content("{\"newName\":\"" + newName + "\"}")
+            .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andExpect(status().isNoContent()).andExpect(content().string(""));
 
+        verify(dataSetService).putRename(oldName, input);
+        verifyNoMoreInteractions(dataSetService);
+    }
+
```
[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java]: added failing tests

File: data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunnerTest.java (hunk 2 above)
Comment (NakulManchanda) [reply data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/PutDataSetRenameZosmfRequestRunnerTest.java]: added more test for coverage

File: data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (stevenhorsman) [reply data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java]: I'm not seeing any new tests here?

[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java]: you are right, added failing test at service level, will add in controller too

[the remark this answers is not in the document]
Comment (NakulManchanda) [reply data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java]: should be good now, added one failing test here

File: data-sets-api-server/src/main/java/org/zowe/data/sets/controller/DataSetsController.java [hunk 7]
```diff
@@ -5,7 +5,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  *
- * Copyright IBM Corporation 2016, 2019
+ * Copyright IBM Corporation 2016, 2019, 2020
```
Comment (stevenhorsman) [inline data-sets-api-server/src/main/java/org/zowe/data/sets/controller/DataSetsController.java]: Copyright is supposed to bookend, so this should be Copyright IBM Corporation 2016, 2020

Comment (NakulManchanda) [reply data-sets-api-server/src/main/java/org/zowe/data/sets/controller/DataSetsController.java]: thanks again, just modified

**Pull request merged**